packages/pharos/src/utils/mixins/pharos-component.ts

```diff
@@ -0,0 +1,24 @@
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
+import type { Constructor } from '@open-wc/dedupe-mixin';
+import type { LitElement } from 'lit';
+
+/**
+ * @param Base The base class for the mixin to modify.
+ * @returns A mixin to set a data attribute with the component class name.
+ */
+const PharosComponentImplementation = <T extends Constructor<LitElement>>(Base: T): T => {
```

Comment:
I think this is the bit we'll want to implement ahead of the breaking change, give people a chance to upgrade to the minor version, and then all the new `[data-pharos-component]` selectors will work across the major version gap.

Reply:
My only concern is it does require folks to update their bundler configs which may technically constitute a breaking change 🤔

Reply:
Can you say more about that? Not sure I follow.

Reply:
Step 5 of the migration guidelines is needed to prevent minification of the class names for that attribute.

Reply:
I see. I think the change itself is non-breaking, but the need to update configurations is strategic and necessary before upgrading to the major version.